src/main/java/pl/javastart/task/Processor.java

```diff
@@ -0,0 +1,45 @@
+package pl.javastart.task;
+
+public class Processor extends Component implements Overclockable {
+    int timing;
+    double currentTemperature;
+    static final int INCRESE_TEMPERATURE = 10;
+    static final int INCRESE_TIMING = 100;
+    static final int MAX_SAFE_TEMPERATURE = 60;
+
+    public void setTiming(int timing) {
+        while (timing - this.timing >= INCRESE_TIMING) {
+            int i = this.timing + INCRESE_TIMING;
+            currentTemperature = currentTemperature + INCRESE_TEMPERATURE;
+            overclock(currentTemperature, i);
+        }
+        this.timing = timing;
+    }
+
+    @Override
+    public void overclock(double currentTemperature, int timing) {
```

Comment:
To logika w metodzie overclock() powinna dbać o wyliczanie tego o ile wzrośnie temperatura. Tutaj albo nie powinno być żadnych parametrów, albo tylko timing